File: Provas/2016.1/questao1-java/Composite.java
import java.util.ArrayList;

public class Composite extends IComponent
{
    public Composite() {
        children = new ArrayList<IComponent>();
    }
    public boolean addChild(IComponent child) {
        children.add(child);
        return true;
    }
    public boolean removeChild(IComponent child) {
        children.remove(child);
        return true;
    }
    public void aumentarPreco(double percentual) {
        for(IComponent child: children)
            child.aumentarPreco(percentual);
    }
    ArrayList<IComponent> children;
}

File: Provas/2016.1/questao1-qt/composite.cpp
#include "composite.h"

#include "icomponent.h"

Composite::Composite()
{

}

bool Composite::addChild(IComponent *child)
{
    m_children << child;
    return true;
}
bool Composite::removeChild(IComponent *child)
{
    m_children.removeAll(child);
    return true;
}
void Composite::aumentarPreco(double percentual)
{
    foreach(IComponent *child, m_children)
        child->aumentarPreco(percentual);
}
